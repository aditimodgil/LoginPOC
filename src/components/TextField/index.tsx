import React from "react";
import { ITextFieldProps } from "../../types";
import { StyledInput } from "../helper";

const TextField = React.forwardRef(
  ({ type, placeholder, name , value}: ITextFieldProps, ref:((instance: HTMLInputElement | null) => void) | React.RefObject<HTMLInputElement> | null | undefined ) => (
    <StyledInput type={type} placeholder={placeholder} name={name} value={value} ref={ref} />
  )
);

export default TextField;
