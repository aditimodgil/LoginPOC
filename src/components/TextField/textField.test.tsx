import React from 'react';
import TextField from './index';
import {shallow} from 'enzyme';

describe('<TextField />', () => {
    const props = {
        type : "text",
        name : "email",
        placeholder : "email" }

 
    it('should render without error', ()=>{
      const wrapper = shallow(<TextField {...props}/>);
      expect(wrapper.find('TextField')).toMatchSnapshot();
    })
})
