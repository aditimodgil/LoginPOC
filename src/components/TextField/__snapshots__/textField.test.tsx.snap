// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render without error 1`] = `ShallowWrapper {}`;
